src/main/java/com/minecolonies/coremod/entity/pathfinding/MinecoloniesAdvancedPathNavigate.java

```diff
@@ -262,7 +267,17 @@ public boolean setPath(@Nullable final Path path, final double speed)
             this.currentPath = null;
             return false;
         }
+        return super.setPath(convertPath(path), speed);
+    }
 
+    /**
+     * Converts the given path to a minecolonies path if needed.
+     *
+     * @param path given path
+     * @return resulting path
+     */
+    private Path convertPath(final Path path)
+    {
         final int pathLength = path.getCurrentPathLength();
         Path tempPath = null;
         if (pathLength > 0 && !(path.getPathPointFromIndex(0) instanceof PathPointExtended))
```

Comment:
Why is there this check twice?

Reply:
its used in two places, pathfollow and setpath

Reply:
Ah, almost skipped the convertPath up there.